fastlane/lanes/build.rb

```diff
@@ -397,4 +411,26 @@ def read_version_from_config
       configuration.attributes['VERSION_SHORT']
     end
   end
+
+  def inject_buildkite_analytics_environment(xctestrun_path:)
+    require 'plist'
+
+    xctestrun = Plist.parse_xml(xctestrun_path)
+    xctestrun['TestConfigurations'].each do |configuration|
+      configuration['TestTargets'].each do |target|
+        TEST_ANALYTICS_ENVIRONMENT.each do |key|
+          value = ENV.fetch(key)
+          next if value.nil?
+
+          target['EnvironmentVariables'][key] = value
+        end
+      end
+    end
+
+    File.write(xctestrun_path, Plist::Emit.dump(xctestrun))
+  end
+
+  def buildkite_ci?
+    ENV.fetch('BUILDKITE', false)
+  end
```

Comment:
You might notice all this logic is duplicated from https://github.com/woocommerce/woocommerce-ios/pull/7490.

I considered moving it to the release toolkit, but decided to first get the analytics to work across all suites to start collecting data.

There is non trivial friction in extracting logic in the toolkit and ship it in the form of Fastlane actions.